### PR TITLE
Disability categories

### PR DESCRIPTION
**Pull request by alaric-dfe**
As per https://trello.com/c/wJ8S8o2b/960-disabilities-in-dfe-reference-data

What should the initial owners and users for the docs be? Will Register and Apply both migrate to using this, taking joint ownership and usership?

### CODE REVIEW

File: lib/dfe/reference_data/equality_and_diversity/disabilities.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+          { name: 'Prefer not to say',
+            opt_out: true,
+            hesa_code: {
+              '2019-2020' => '98',
+              '2020-2021' => '98',
+              '2021-2022' => '98',
+              '2022-2023' => '98'
+            } } },
+        schema: DISABILITIES_SCHEMA,
+        list_description: 'Disabilitie classifications',
```
Comment (frankieroberto) [inline lib/dfe/reference_data/equality_and_diversity/disabilities.rb:99]: typo
```suggestion
        list_description: 'Disabilities classifications',
```

File: docs/lists_equality_and_diversity.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+## Equality and Diversity
+
+### `DfE::ReferenceData::EqualityAndDiversity::DISABILITIES`
```
Comment (frankieroberto) [inline docs/lists_equality_and_diversity.md:3]: I think we should refer to this list as "disabilities and health conditions" everywhere, even if that’s more longwinded, as that is how we now refer to the list on Apply, and some of the conditions may not be considered disabilities by some people.

```suggestion
### `DfE::ReferenceData::EqualityAndDiversity::DISABILITIES_AND_HEALTH_CONDITIONS`
```

File: docs/lists_equality_and_diversity.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+```ruby
+require 'dfe/reference_data/equality_and_diversity'
+```
+
+Disability classifications.
```
Comment (frankieroberto) [inline docs/lists_equality_and_diversity.md:9]:
```suggestion
Disabilities and health conditions
```

File: docs/lists_equality_and_diversity.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+These items are intended to be referenced from a multiple-value field. The opt-out option is exclusive; if used, no other values can be used. The absence of any options in the multiple-value field is to be interpreted as the user having no disabilities.
```
Comment (frankieroberto) [inline docs/lists_equality_and_diversity.md:11]:
```suggestion
These items are intended to be referenced from a multiple-value field. The opt-out option is exclusive; if used, no other values can be used. The absence of any options in the multiple-value field is to be interpreted as the user having no disabilities or health conditions affecting daily life.
```

File: docs/lists_equality_and_diversity.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+Owner: Register team.
```
Comment (frankieroberto) [inline docs/lists_equality_and_diversity.md:13]: joint owners?
```suggestion
Owner: Apply and Register teams.
```

File: docs/lists_equality_and_diversity.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+|-------------|----------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `id`        | UUID                                         | A unique identifier for this disability option                                                                                                                 |
+| `name`      | string                                       | The long name of the disability option.                                                                                                                        |
+| `hint`      | optional string                              | An optional hint to be shown alongside the name to users                                                                                                       |
+| `generic`   | optional boolean                             | If present and true, indicates that this is a generic "other" option and, if selected, the user should be prompted for free text to specify any further detail |
```
Comment (frankieroberto) [inline docs/lists_equality_and_diversity.md:26]:
```suggestion
| `generic`   | optional boolean                             | If present and true, indicates that this is a generic "other" option and, if selected, the user should be prompted for free text to optionally specify any further detail |
```

File: docs/lists_equality_and_diversity.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+| `opt_out`   | optional boolean                             | If present and true, indicates that the user has opted out of answering the question. This value is exclusive and cannot be combined with any other option     |
```
Comment (frankieroberto) [inline docs/lists_equality_and_diversity.md:27]: Should we use the same language here as the interface? "opt out" feels subtly different from "prefer not to say" and slightly jargony?

```suggestion
| `prefer_not_to_say`   | optional boolean                             | If present and true, indicates that the user has selected that they prefer not to answer the question. This value is exclusive and cannot be combined with any other option     |
```

File: lib/dfe/reference_data/equality_and_diversity/disabilities.rb (hunk 1 above)
Comment (edwardhorsford) [inline lib/dfe/reference_data/equality_and_diversity/disabilities.rb:99]: Typo

File: lib/dfe/reference_data/equality_and_diversity/disabilities.rb [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+      }.freeze
+
+      DISABILITIES = DfE::ReferenceData::HardcodedReferenceList.new(
+        { 'da4faa34-3851-4e04-959a-92ebea3c2b98' =>
+          { name: 'Autistic spectrum condition or another condition affecting speech, language, communication or social skills',
+            hesa_code: {
+              '2019-2020' => '53',
+              '2020-2021' => '53',
+              '2021-2022' => '53',
+              '2022-2023' => '53'
```
Comment (edwardhorsford) [inline lib/dfe/reference_data/equality_and_diversity/disabilities.rb:22]: I'm unsure we'd need these scoped by year like this - but we can discuss.

**Review by frankieroberto — COMMENTED** [review]
Looks good.

Should we add a note somewhere to say that the exact phrasing of the options has varied over time, whilst remaining broadly consistent and comparable?

eg HESA changed "Blind or a serious visual impairment uncorrected by glasses" to "Blind or have a visual impairment uncorrected by glasses" and on Apply we use "Blindness or a visual impairment not corrected by glasses".

We may make further tweaks to the wording in future, but any significant changes would use a new code?

File: lib/dfe/reference_data/equality_and_diversity/disabilities.rb [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+            } },
+          'd3f0f6de-b9be-4299-ade0-b40eef5d9ef2' =>
+          { name: 'Prefer not to say',
+            opt_out: true,
+            hesa_code: {
+              '2019-2020' => '98',
+              '2020-2021' => '98',
+              '2021-2022' => '98',
+              '2022-2023' => '98'
+            } } },
```
Comment (edwardhorsford) [inline lib/dfe/reference_data/equality_and_diversity/disabilities.rb:97]: I wonder if this should be grouped separately. Certainly in our UI's we'll handle them separately.

File: lib/dfe/reference_data/equality_and_diversity/disabilities.rb [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+        schema: DISABILITIES_SCHEMA,
+        list_description: 'Disabilitie classifications',
+        list_usage_guidance: 'These items are intended to be referenced from a multiple-value field. The opt-out option is exclusive; if used, no other values can be used. The absence of any options in the multiple-value field is to be interpreted as the user having no disabilities.',
```
Comment (edwardhorsford) [inline lib/dfe/reference_data/equality_and_diversity/disabilities.rb:100]: What's the thinking for not having an explicit 'no disabilities' option?

Reasons I think we might want one:

* HESA has a code for it (`95`) that we'll need to be able to map and store.
* Feels less explicit

**Review by edwardhorsford — COMMENTED** [review]
Looking good so far! Would be good to discuss a few bits.

**Comment by claire-hughez** [comment]
_What should the initial owners and users for the docs be? Will Register and Apply both migrate to using this, taking joint ownership and usership?_

We'll be prioritising Register using it, but only really works if Apply does too? @MylesJarvis 

In terms of ownership, weirdly, kinda HESA owns it hahaha... Do we want to agree we move to what HESA does? Or something else? Candidate needs seem to trump here though, so also happy to defer to Apply.

**Comment by edwardhorsford** [comment]
I think the owners should be Apply and Register in the first instance.

**Comment by frankieroberto** [comment]
Agree that owners should be Apply and Register. We currently align with HESA, and may continue to do so, but we might also diverge (on more than just minor wording) in future.

We _could_ also include codes that map to the [Impairment harmonised standard](https://analysisfunction.civilservice.gov.uk/policy-store/impairment/) from the Government Analysis Function, although that standard is also currently being reviewed.

File: lib/dfe/reference_data/equality_and_diversity/disabilities.rb (hunk 10 above)
Comment (alaric-dfe) [reply lib/dfe/reference_data/equality_and_diversity/disabilities.rb]: Ok, I've added one, and rearranged how special values are denoted to simplify it a bit now there's three

File: lib/dfe/reference_data/equality_and_diversity/disabilities.rb [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+        id: :string,
+        name: :string,
+        hint: { kind: :optional, schema: :string },
+        hesa_code: { kind: :code, pattern: /^[0-9]{2}$/ },
+        kind: { kind: :optional, schema: :symbol }
+      }.freeze
+
+      DISABILITIES_AND_HEALTH_CONDITIONS = DfE::ReferenceData::HardcodedReferenceList.new(
+        { 'da4faa34-3851-4e04-959a-92ebea3c2b98' =>
+          { name: 'Autistic spectrum condition or another condition affecting speech, language, communication or social skills',
```
Comment (edwardhorsford) [inline lib/dfe/reference_data/equality_and_diversity/disabilities.rb:14]: Indentation

**Pull request merged**